packages/task/src/browser/task-service.ts

```diff
@@ -648,7 +689,7 @@ export class TaskService implements TaskConfigurationClient {
             TERMINAL_WIDGET_FACTORY_ID,
             <TerminalWidgetFactoryOptions>{
                 created: new Date().toString(),
```

Comment:
this code for long time bother me: `TerminalWidgetFactoryOptions` === terminal id, i.e. `TerminalWidgetFactoryOptions` should be a minimal amount of attributes uniquely identifying an instance, e.g. for and editor, it is a uri without a fragment.

I cannot get why `created` and `destroyTermOnClose` should be a part of a terminal id.

Reply:
`created` and `destroyTermOnClose` are not part of a terminal id. As far as I can tell they are part of the `TerminalWidgetFactoryOptions`.

they were added long time ago in this commit 
```
  Sven Efftinge, 2 years ago   (September 17th, 2017 10:48am)

[storage] layout preservation and storage service implementation

Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

c7c46de ≪ ≡ ↗ ⋯

Changes   —   75a48a9  ⟷  c7c46de
```
I will take a look at the commit and see what I can do in my future change

Reply:
i looked into the commit where Sven introduce the interface: https://github.com/eclipse-theia/theia/commit/c7c46de3e6b4f4f9d1d94e8efb2081ed916a8afc#diff-bfac6f937a5be80c608b3b3ad2f865bdR31

My guess is that using the time stamp on the `created` property is probably  the easiest solution to identify terminals, because in Theia users are free to create as many terminal widget as they want. 

In my opinion `TerminalWidgetFactoryOptions` contains an ID but it is not used just as an ID.
```
interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOptions> {
    created: string
}
```
As you can see in the line you added your comment, it is used as an object to pass "options" into the widget manager, so that the widget manager had sufficient information to create the terminal widget.

Reply:
ok